README.md

```diff
@@ -1,25 +1,120 @@
+applies some rendering instructions (styple) to the input map, using the [Lua](https://www.lua.org)
+scripting language. The rendering definition can be based on shapes, typically polygons, that
+define areas of the map that must be rendered specifically (for example to highlight the
+buildings of your laboratory in a campus). Shapes must be built with a tool like
+[QGIS](https://www.qgis.org/fr) and stores in a [shapefile](https://wiki.openstreetmap.org/wiki/Shapefiles).
+
+This repository provides a Dockerfile and configuration files for `tileserver` and `tilemaker`.
+These files are based on CERN configuration and can be used as templates to build your own
+configuration. The documentation below describes the main parameters that need to be customized.
+
+## Building the tileserver container
+
+### Dockerfile
+
+The Dockerfile is made up of two stages: each stage produces a different container.
+* the first one downloads the [OpenStreetMap data](https://planet.osm.ch/), crops it to a bounding box and generates an `*.mbtiles` file containing relevant data;
+  * Relies on [osmconvert](https://wiki.openstreetmap.org/wiki/Osmconvert) tool to extract the region
+of interest from an OSM map.
+* the second one sets up a lightweight tileserver-gl using the data generated above;
 
-## Usage
+### Shapefile
+
+### Styles
+
+### Build command
+
+As for any Docker container that you want to build from a Dockerfile, go into the directory
+containing the Dockerfile and run the following command:
 
 ```
 $ docker build . -t indico-maps
-$ docker run -p 8080:8080 -it indico-maps
 ```
 
-Then, just connect to https://yourhost:8080.
+''Note: `indico-maps` will be the tag of (name referring to) the container. You can use any
+name you prefer: nothing in the configuration refers to this name.`
+
+## Running the tile server
+
+### tileserver configuration
+
+The `tileserver` configuration file is located in `tileserver/config.json` in the repository.
+It contains 2 dictionaries which have normally one key each. The key must be the same in both
+dictionaries and will be your site name (used in the tile server URL configured in Indico).
+
+The 2 dictionaries are:
+* `data`: it contains an entry `mbtiles` which is the name of the file that contains the map
+tiles. It must match the name of the file produced by `tilemaker`.
+* `styles`: this dictionary has 2 entries:
+  * `styles`: must reflect you style configuration, i.e. the `config.json` path relative to the
+  `styles` directory.
+  * `tilejson/bounds`: an array of GPS coordinates of your site OSM map (processed by tilemaker)
+  for the south-west and the north-east corners with `latitude` the `longitude` for each one.
+
+### Starting the tileserver
+
+The typical command to run the tileserver container built previously is:
+
+```
+$  docker run -dit --restart unless-stopped -p 8080:8080 indico-maps
+```
+
+''Note: if you use a container tag different from `indico-maps` when you built the container,
```

Comment:
I think it's an "image tag" (you create a container from an image - building only creates an image. not a container)

Reply:
is the `''` at the beginning supposed to be there?